### PR TITLE
Makes all default job titles, defines.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Not sure if I should make the id trim also use this define, or if I should expand this to even more jobs (like CC official), but I did this all this for now, to wait for feedback

@LemonInTheDark i did it

## Why It's Good For The Game

Game?

## Changelog

/🆑 
fix: Geneticists are now considered Science in terms of the Borg petting mood buff.
🆑

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm [hunk 1]
```diff
@@ -301,7 +301,7 @@
 	H.dna.add_mutation(DWARFISM)
 
 /obj/effect/mob_spawn/human/corpse/damaged/legioninfested/Initialize(mapload)
-	var/type = pick_weight(list("Miner" = 66, "Ashwalker" = 10, "Golem" = 10,"Clown" = 10, pick(list("Shadow", "YeOlde","Operative", "Cultist")) = 4))
+	var/type = pick_weight(list("Miner" = 66, "Ashwalker" = 10, "Golem" = 10, "Clown" = 10, pick(list("Shadow", "YeOlde","Operative", "Cultist")) = 4))
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:304]: giving a merge conflict to armhulen smh

Comment (JohnFulpWillard) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:304]: I painfully went through every single mention of every single job.

File: code/datums/id_trim/jobs.dm [hunk 2]
```diff
@@ -17,7 +17,7 @@
 	var/list/minimal_wildcard_access = list()
 	/// Static list. Cache of any mapping config job changes.
 	var/static/list/job_changes
-	/// What config entry relates to this job. Should be a lowercase job name with underscores for spaces, eg "prisoner" "research_director" "head_of_security"
+	/// What config entry relates to this job.
```
Comment (MrMelbert) [inline code/datums/id_trim/jobs.dm:20]: the map config entries may need to be updated to accommodate this change

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Thank you john, you've done a good deed today

File: code/datums/id_trim/jobs.dm [hunk 3]
```diff
@@ -83,7 +83,7 @@
 	trim_state = "trim_assistant"
 	extra_access = list(ACCESS_MAINT_TUNNELS)
 	minimal_access = list()
-	config_job = "assistant"
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/datums/id_trim/jobs.dm:86]: I cannot, for the life of me, figure out what the hell this is supposed to do.

**Comment by JohnFulpWillard** [comment]
I literally deleted config_job and it doesnt seem like anything was affected??
image

File: code/datums/id_trim/jobs.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/datums/id_trim/jobs.dm:86]: It lets mappers add job specific changes to their map.json files
image
stuff like this.

Should prob just be left alone

[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/datums/id_trim/jobs.dm]: done

File: code/controllers/subsystem/job.dm [hunk 4]
```diff
@@ -787,16 +788,30 @@ SUBSYSTEM_DEF(job)
...
+		JOB_CHIEF_ENGINEER,
+		JOB_CHIEF_MEDICAL_OFFICER,
+	)
 
-	additional_jobs_with_icons = list("Emergency Response Team Commander", "Security Response Officer", "Engineering Response Officer", "Medical Response Officer", \
-		"Entertainment Response Officer", "Religious Response Officer", "Janitorial Response Officer", "Death Commando", "Security Officer (Engineering)", \
-		"Security Officer (Cargo)", "Security Officer (Medical)", "Security Officer (Science)")
+	additional_jobs_with_icons = list(
+		"Emergency Response Team Commander", "Security Response Officer", "Engineering Response Officer", "Medical Response Officer", \
+		"Entertainment Response Officer", "Religious Response Officer", "Janitorial Response Officer", "Death Commando", \
```
Comment (LemonInTheDark) [inline code/controllers/subsystem/job.dm:812]: Should these have defines?

File: code/game/machinery/computer/security.dm [hunk 5]
```diff
@@ -905,7 +905,7 @@ What a mess.*/
 							temp += "<li><a href='?src=[REF(src)];choice=Change Criminal Status;criminal2=released'>Discharged</a></li>"
 							temp += "</ul>"
 					if("rank")
-						var/list/L = list( "Head of Personnel", "Captain", "AI", "Central Command" )
+						var/list/L = list(JOB_CAPTAIN, JOB_HEAD_OF_PERSONNEL, JOB_AI, "Central Command")
```
Comment (LemonInTheDark) [inline code/game/machinery/computer/security.dm:908]: this is also a bit weird

File: code/game/machinery/computer/crew.dm [hunk 6]
```diff
@@ -150,7 +150,7 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
 		"Emergency Response Team Commander" = 220,
 		"Security Response Officer" = 221,
 		"Engineer Response Officer" = 222,
-		"Medical Response Officer" = 223
```
Comment (LemonInTheDark) [inline code/game/machinery/computer/crew.dm:153]: maybe just go off this list?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Still very happy with this, seems like a few "job title" like uses are not converted, but I have a feeling that's because they're tied to jobs without datums

**Comment by LemonInTheDark** [comment]
> I literally deleted config_job and it doesnt seem like anything was affected?? 

Cursed by poorly documented vars

File: code/controllers/subsystem/job.dm (hunk 4 above)
Comment (JohnFulpWillard) [reply code/controllers/subsystem/job.dm]: they do now

File: code/game/machinery/computer/crew.dm (hunk 6 above)
Comment (JohnFulpWillard) [reply code/game/machinery/computer/crew.dm:153]: I hate this list, ruined my whole life. I don't see why the jobs displayed on the crew monitor would be hardcoded onto it.

**Pull request merged**